libensemble/gen_funcs/persistent_pds.py

```diff
@@ -0,0 +1,165 @@
+import numpy as np
+import numpy.linalg as la
+import scipy.sparse as spp
+
+from libensemble.message_numbers import STOP_TAG, PERSIS_STOP, FINISHED_PERSISTENT_GEN_TAG
```

Comment:
In the other gens, libE is able to tell the gen to stop by sending `STOP_TAG` or `PERSIS_STOP`. Can you please adjust this gen to do the same.